hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -113,7 +118,8 @@ public HoodieFileGroupReader(HoodieReaderContext<T> readerContext,
                                Schema avroSchema,
                                TypedProperties props,
                                long start,
-                               long length) {
+                               long length,
+                               boolean positiionBasedWriteEnabled) {
```

Comment:
Sounds confusing to have `...writeEnabled` in a reader class. Maybe rename to `shouldUseRecordPositions`?